src/redux/reducers/viewports.js

```diff
@@ -48,6 +50,28 @@ const viewports = (state = defaultState, action) => {
         action.data
       )
 
+      return Object.assign({}, state, { viewportSpecificData })
+    case SET_VIEWPORT_SPECIFIC_DATA:
+      currentData =
+        cloneDeep(state.viewportSpecificData[action.viewportIndex]) || {}
+      viewportSpecificData = cloneDeep(state.viewportSpecificData)
+      viewportSpecificData[action.viewportIndex] = merge(
+        {},
+        currentData,
+        action.data
+      )
+
+      return Object.assign({}, state, { viewportSpecificData })
+    case 'SET_ACTIVE_VIEWPORT_SPECIFIC_DATA':
```

Comment:
- Can we add this to ActionTypes?
- Can we chain actions? IE. this calls `SET_VIEWPORT_SPECIFIC_DATA`, but passes the `activeViewportIndex`?

Reply:
done